pkg/apis/pdapi/pdapi.go

```diff
@@ -0,0 +1,797 @@
+// Copyright 2018 PingCAP, Inc.
```

Comment:
`pkg/apis` is a directory to store Kubernetes API types. IMO it's not a good idea to put PD controller client here.

Reply:
Where would a good place be for this?
I don't think of this as a PD controller but instead just the PD API interaction. Logic to "control" PD is elsewhere. The code take types from this project, serializes them for the API, sends a request, and then convert the response into a type for this project. The file is renamed now to pdapi. There is an interface PDControlInterface that I would also rename, but I tried to minimize my changes to only what is required.